intranet/apps/auth/views.py

```diff
@@ -183,6 +185,21 @@ def post(self, request):
 
             default_next_page = "index"
 
+            if request.user.is_student:
+                if settings.ENABLE_PRE_EIGHTH_REDIRECT:
+                    # Redirect to eighth signup page if the user isn't signed up for eighth period activities
+                    now = timezone.localtime(timezone.now())
+
+                    today_blocks = EighthBlock.objects.filter(date=now.date())
+                    if today_blocks.exists():
+                        for block in today_blocks:
+                            block_start = timezone.make_aware(datetime.combine(now.date(), block.signup_time), now.tzinfo)
+                            time_difference = block_start - now
+                            if timedelta(0) <= time_difference <= timedelta(minutes=10):
```

Comment:
```suggestion
                            if timedelta(0) <= time_difference <= timedelta(minutes=20):
```
If this happens, we probably want a more significant time period before signups close.